src/x-server-local.c

```diff
@@ -481,8 +481,10 @@ x_server_local_start (DisplayServer *display_server)
     if (server->priv->vt >= 0)
         g_string_append_printf (command, " vt%d -novtswitch", server->priv->vt);
 
-    if (server->priv->background)
-        g_string_append_printf (command, " -background %s", server->priv->background);
+    if (!server->priv->background)
+        x_server_local_set_background (server, "none");
+
+    g_string_append_printf (command, " -background %s", server->priv->background);
```

Comment:
Right, it seems nothing actually sets the background. I think in Ubuntu we default to none background so I haven't noticed this.

The background shouldn't be set in this method - we should either set the background when we make the object or set the default to "none" (in x_server_local_init).

I think this feature wasn't always available in X servers, but it's probably safe to assume it is. I think there's other features we assume anyway.

Can you put this into a separate PR with a better commit description.